http-clients/apache5-client/src/main/java/software/amazon/awssdk/http/apache5/Apache5HttpClient.java

```diff
@@ -769,8 +769,8 @@ private static ConnectionConfig getConnectionConfig(AttributeMap standardOptions
                                 .setSocketTimeout(Timeout.ofMilliseconds(
                                     standardOptions.get(SdkHttpConfigurationOption.READ_TIMEOUT).toMillis()));
             Duration connectionTtl = standardOptions.get(SdkHttpConfigurationOption.CONNECTION_TIME_TO_LIVE);
-            if (!connectionTtl.isZero()) {
-                // Skip TTL=0 to maintain backward compatibility (infinite in 4.x vs immediate expiration in 5.x)
+            if (!connectionTtl.isNegative()) {
```

Comment:
Note: using `isNegative()` rather than `isPositive()` allows customer to have access to immediate expiry by directly setting `Duration.ZERO`. Should we allow this?